Logger/FullName.cs

```diff
@@ -0,0 +1,15 @@
+﻿namespace Logger
+{
+    //The FullName is a value type that could be modified through copies when called upon
+    //while keeping the main copy unchanged.
+    //The FullName record will be immutable since the changes will be performed on copies.
+    public record struct FullName(string FirstName, string LastName, string? MiddleName = null)
```

Comment:
```suggestion
    public readonly record struct FullName(string FirstName, string LastName, string? MiddleName = null)
```
Our instructor recommends always making structs readonly. VS also recommends setting it to readonly since it's all getters anyway.